sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeFormatterHelper.scala

```diff
@@ -227,6 +227,11 @@ private object DateTimeFormatterHelper {
     formatter.format(LocalDate.of(2000, 1, 1)) == "1 1"
   }
   final val unsupportedLetters = Set('A', 'c', 'e', 'n', 'N', 'p')
+  // SPARK-31892: The week-based date fields are rarely used and really confusing for parsing values
+  // to timestamp, especially when they are mixed with other non-week-based ones. we have tried our
+  // best to restore the behavior change between 2.4 and 3.0 and failed, see
+  // https://github.com/apache/spark/pull/28674
```

Comment:
We usually don't link PRs in the code comment. I think we can remove "We have tried our ...."